src/lib.rs

```diff
@@ -174,6 +174,7 @@
     missing_debug_implementations
 )]
 #![allow(clippy::type_complexity)]
+#![allow(clippy::field_reassign_with_default)]
```

Comment:
I tried to address these warnings but I found the less code less readable afterwards.

Reply:
Hmm, this lint _should_ be a fairly good one. What did you find problematic about it?
Also, I think we can remove the `allow(clippy::type_complexity)`, since that lint should now be off by default anyway.

Reply:
Sorry for the delay. Regarding `allow(clippy::field_reassign_with_default)`, we're getting warnings in:
1. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L834-L836
2. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L846-L853
3. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1004-L1019
4. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1099-L1111
5. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1439-L1440
6. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1478-L1479
7. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1519-L1520
8. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1569-L1570
9. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1619-L1620

The ones I found awkward were 3 and 4, but probably due to the `self.make_placement` request. I'm now thinking that if that request is performed before the initialization, then these changes would be fine. What do you think?

As for `allow(clippy::type_complexity)`, if we remove it, we get a warning for example in: https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/lib.rs#L335-L337

Reply:
I went ahead and did it in 9229fd515310660c9bbbe1b2303d1e00d691fe8a.